server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -505,6 +507,10 @@ public static APIBlock readFrom(StreamInput input) throws IOException {
 
     private final AutoExpandReplicas autoExpandReplicas;
 
+    private final boolean isSearchableSnapshot;
+
+    private final boolean isPartialSearchableSnapshot;
```

Comment:
NIT: Looks like there are quiet a number of fields already. Would it maker sense to merge above into enum or something?

Reply:
Yea that is a fair point. I'd rather not change the logic here yet though. It's quite complicated with the two booleans in various spots and I think we should keep this kind of change as trivial as possible.
++ though to trying to find ways of reducing complexity in this class in a follow-up, I'll think about it